src/commands/commerce/extension/map.ts

```diff
@@ -147,4 +138,38 @@ export class MapExtension extends SfdxCommand {
             return returnResult;
         }
     }
+    private deleteDuplicateMaps(extensionName: string, userName: string, storeid: string): void {
+        let epnVal: string;
+        const EPNQuery = forceDataSoql(
+            `SELECT ExtensionPointName FROM RegisteredExternalService WHERE DeveloperName='${extensionName}'`,
+            userName
+        );
+        if (EPNQuery !== undefined && EPNQuery.result !== undefined && EPNQuery.result.totalSize > 0) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+            epnVal = EPNQuery.result.records[0]['ExtensionPointName'];
+        } else {
+            throw new SfdxError(msgs.getMessage('extension.map.errorEPN'));
+        }
+
+        const existingIds = forceDataSoql(
+            `SELECT DeveloperName,ExternalServiceProviderType FROM RegisteredExternalService WHERE ExtensionPointName='${epnVal}' AND ExternalServiceProviderType='Extension'`,
+            userName
+        );
+        for (const element of existingIds.result.records) {
+            const id = (element['ExternalServiceProviderType'] as string)
+                .concat('__' as string)
+                .concat(element['DeveloperName'] as string);
+            try {
+                const deleteId = forceDataSoql(
+                    `SELECT Id FROM StoreIntegratedService WHERE Integration='${id}' AND StoreId='${storeid}'`,
+                    userName
+                ).result.records[0].Id;
+                if (deleteId !== undefined) {
+                    this.ux.log(msgs.getMessage('extension.map.previousEPN', [`'${id}'`, 'due to duplicate EPN']));
+                    forceDataRecordDelete('StoreIntegratedService', deleteId, this.org.getUsername(), 'ignore');
+                }
+                // eslint-disable-next-line no-empty
+            } catch (e) {}
```

Comment:
with the condition on line 167 are we able to remove the try catch now?

Reply:
It still breaks the command and exits the code if there is no catch